applications/sintering/include/pf-applications/sintering/operator_sintering_base.h

```diff
@@ -178,26 +178,29 @@ namespace Sintering
                   temp[counter++] = etaijSum;
                 }
 
+              const auto free_energy_eval =
+                free_energy.template eval<EnergyZero, n_grains>(val);
```

Comment:
Is everything threadsafe?

Reply:
Yes, with this solution everything is threadsafe, since `FreeEnergy::Evaluation` object is always instantiated locally, precomputes some members (if needed), and then only read operations are executed.